contracts/deployables/counter-sign/CounterSignV1.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.30;
+
+import {ICounterSignV1} from "../../interfaces/decent/deployables/ICounterSignV1.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/*
+Design discussion: https://docs.google.com/document/d/1xp8w25O7CnsMf3cVPX_AZ3XT0k6DaAaWWl50p-VCj9A/edit?usp=sharing
+*/
+
+contract CounterSignV1 is Initializable, ICounterSignV1 {
+    string private agreementUrl;
+    uint256 private expirationTimestamp;
+    bool private executed;
+
+    address[] private signers;
+    mapping(address => string) private entityNames;
+    mapping(address => string) private signerPersonalNames;
+    mapping(address => bool) public signedAddresses;
+
```

Comment:
This comment and the next few all have to do with removing the for loops in `getIsExecutable` and `_isSigner`, which could get expensive if there are many signers.

Start with introducing two new variables:

```sol
mapping(address => bool) private isSignerAddress;
uint256 private signedCount;
```